app/models/wpcc/contributions_calendar.rb

```diff
@@ -0,0 +1,56 @@
+module Wpcc
+  class ContributionsCalendar
+    attr_reader :eligible_salary,
+                :salary_frequency,
+                :employee_percent,
+                :employer_percent
+
+    PERIODS_FILE = Wpcc::Engine.root.join('config', 'contribution_periods.yml')
+    PERIODS = YAML.load_file(PERIODS_FILE)
+
+    def initialize(eligible_salary:,
+                   employee_percent:,
+                   employer_percent:,
+                   salary_frequency:)
+
+      @eligible_salary = eligible_salary
+      @employee_percent = employee_percent
+      @employer_percent = employer_percent
+      @salary_frequency = salary_frequency
+    end
+
+    def schedule
```

Comment:
Yeah, as long as there's an easy way of presenting the data to the views rather than as a giant bulk without much context. Hopefully, having suitably named presenter methods will help the _Frontend_ with consuming the API.